Automattic/wp-calypso
# Jetpack Settings: When saving, skip settings of disabled modules

This PR addresses an issue where we can't save any settings form that contains Jetpack settings and at least one disabled module. The reason for that is because the `/settings` endpoint does not allow settings to be updated for disabled modules. So, when we try saving with at least one controlled module disabled, we were receiving an error notice. So this PR attempts to fix this by omitting the settings of disabled modules upon saving.

The downsides of this approach is that some boilerplate is necessary to achieve all of this:

* Include the setting that contains the name of the module in the list of controlled fields that we pass to `wrapSettingsForm`
* Include the list of modules that we manage and their settings that we manage in `filterSettingsByActiveModules()`, so it knows how and when to filter the settings when the corresponding module is disabled.

Also, we'll need to make sure that this works properly for all current and future Jetpack Settings cards.

To test:
1. Checkout this branch
2. Go to /settings/writing
3. Disable one of the modules in the Theme Enhancements card and play with the settings of the other one.
4. Click any "Save Changes" button and verify saving works as expected and a success message is displayed.
5. Try steps 3-4 by disabling the other module in Theme Enhancements.
6. Try testing the same with any Jetpack Settings settings/cards in the Discussion tab.

Fixes #10950.
